src/containers/WatchActivities.tsx

```diff
@@ -0,0 +1,85 @@
+import usePubSub from '@hooks/usePubSub'
+import { ayonApi } from '@/services/ayon'
+import { FC } from 'react'
+import { useDispatch, useStore } from 'react-redux'
+import { $Any } from '@/types'
+type ActivityMessage = {
+  [key: string]: any
+
+  summary:
+    | {
+        activity_id: string
+        activity_type: string
+        references: {
+          entity_id: string
+          entity_type: string
+          reference_type: string
+        }[]
+      }
+    | undefined
+    | null
+}
+
+interface WatchActivitiesProps {}
+
+const WatchActivities: FC<WatchActivitiesProps> = ({}) => {
+  const dispatch = useDispatch()
+
+  const store = useStore()
+
+  //   subscribe to inbox.message topic
+  usePubSub(
+    'activity',
+    (topic: string, message: ActivityMessage) => {
+      const references = message.summary?.references
+      const entityIds = references?.map((reference) => reference.entity_id) || []
```

Comment:
I think we should these two lines in else condition because we are not using this in if condition, so why to perform map function for if condition, it could be a expensive operation, so can we please move these two lines in else condition?

Reply:
Agree from a readability and formating perspective so I have moved.

It would have negligible affect on performance as references would only ever be 1-20ish items. Maybe if it was 1000+ plus items.

Reply:
Oh understood, I thought references could be 1000+